### ISSUE TITLE
Cannot Get the Same Result Picture as Samples

### ISSUE THREAD

**Issue by yjump** [pshegde96/painter-nn#1]
Hi Parikshit Hegde,
     Sorry to bother you. I am learning to use tensorflow by using your painter-nn project recently. But I cannot get the same picture as samples which I guess l2 loss is less than 1200? I would like to know if there is some skills to train this full-connect NN?
     I have tried to adjust bacthsize, optimizer type or add layers. But even I use 100 hlayers to train the net still cannot get a result as your sample. This makes me very confused. Did I miss anything important?
    I add two pictures of my results below.
    Thank you a lot for your project and assistance.

**Comment by yjump** [comment]
![image](https://cloud.githubusercontent.com/assets/26101847/24034311/c96ac852-0b2b-11e7-9ffd-3f8ab0af2991.png)

**Comment by pshegde96** [comment]
Hi,
The samples that I have put in the samples/ directory are all with the default parameter settings that I have put in the painter.py file. It has 6 hidden layers with each layer containing 20 neurons with a ReLU activation after each layer. I have set a batch size of 1000 and used the default parameters for the Adam Optimizer. Please try running the programs with the these default settings and see if you are still not able reproduce the results. I think you should be able to reproduce these results.

Coming to your particular comments about the earth.jpg sample. I feel that 100 layers is way to many. Higher number of layers is not always better. With such a high number of layers, the gradients that propagate to the layers in the beginning of the network will be very small, and hence the network will not learn very well. That is why it looks like your network hasn't been able to learn well even after 5000 epochs. Experiment with number of layers upto 10.

Also, regarding the l2 loss. The earth sample that was generated by the network has an l2 loss of around               of around 300. A sample with l2 loss of 1200 will not be of a very good quality.

**Comment by yjump** [comment]
Hi, Thanks a lot!
I really appreciate your kind insights on my last comment! I have been working on your project as a game for half a week, and tried many settings including your default setting. It did not work very well， listed below (6 hlayers, 4000 epoch, Adam optimizer and 1000 batch size):
![image](https://cloud.githubusercontent.com/assets/26101847/24067991/0724eb4e-0bbf-11e7-9345-9a381e0712ed.png)
I also read Karpathy's convNetJS demo source code ( also a 7 hlayer and 20 neurons simple model), but the samples I got are really of a bad quality. 

I know too more layers may make gradients disappear ( this full-connected NN is not ResNet ). However, the fact shows that the loss still improved as the number of layers grows up . I was surprised but practice speaks louder than words. I think the reason is the usage of ReLU can help the gradients propagate.

The best loss result I got was around 450， using 100 layers and a 80 neurons with ReLU.

I think the different running environment may lead to different results. I am using python 3.3 + tensorflow 1.0 + win 10 x64. I will try this project in a more reliable  environment and see how it works.

Anyway, thank you for your kind response. Hope you a good weekend~
